### PR TITLE
Environment.GetFolderPath: handle case of an unknown user

### PR DESCRIPTION
**Pull request by tmds**
The call to Persistentfiles.GetHomeDirectory may return null when
HOME is unset and there is no entry for the user in /etc/passwd.
This null-case was not covered.

CC @stephentoub @danmosemsft

### CODE REVIEW

File: src/System.Runtime.Extensions/src/System/Environment.Unix.cs [hunk 1]
```diff
@@ -112,17 +112,19 @@ private static string GetFolderPathCoreWithoutValidation(SpecialFolder folder)
...
-            string home;
+            string home = null;
             try
             {
                 home = PersistedFiles.GetHomeDirectory();
             }
             catch (Exception exc)
             {
                 Debug.Fail($"Unable to get home directory: {exc}");
+            }
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/src/System/Environment.Unix.cs]: Nit: can you add a newline after the closing brace?

**Comment by stephentoub** [comment]
I assume this fixes https://github.com/dotnet/corefx/issues/32716?

Is there a way to have an automated test for this? (Presumably not very easily.)

**Comment by danmosemsft** [comment]
@MattGal which of our CI and official build runs if any run as root? Can we sudo in a test?

**Comment by MattGal** [comment]
I believe neither builds nor tests run as root today (helix-bot or dotnet-bot depending on the machine), but sudo from either should be fine.  Ping me offline for details.

**Pull request merged**